File: problemas/ad-hoc/2377/2377.cs
using System;
using System.Collections.Generic;
using System.Linq;

class URI
{
    static void Main(string[] args)
    {
        List<int> LD = Console.ReadLine().Trim().Split(' ').Select(x => int.Parse(x)).ToList();
        List<int> KP = Console.ReadLine().Trim().Split(' ').Select(x => int.Parse(x)).ToList();
        int L = LD[0];
        int D = LD[1];
        int K = KP[0];
        int P = KP[1];

        Console.WriteLine(L * K + (L / D) * P);
    }
}

File: problemas/ad-hoc/2377/2377.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
  public static void main(String[] args) throws IOException {
    InputStreamReader ir = new InputStreamReader(System.in);
    BufferedReader in = new BufferedReader(ir);

    String[] numeros = in.readLine().trim().split(" ");
    int l = Integer.parseInt(numeros[0]);
    int d = Integer.parseInt(numeros[1]);

    numeros = in.readLine().trim().split(" ");
    int k = Integer.parseInt(numeros[0]);
    int p = Integer.parseInt(numeros[1]);

    System.out.println(l * k + (l / d) * p);
  }
}